libpromises/dbm_migration.c

```diff
@@ -64,10 +65,46 @@ bool DBMigrate(DBHandle *db, dbid id)
         {
             if (step_version == DBVersion(db))
             {
-                if (!(*step)(db))
+                /* Create a backup */
```

Comment:
Isn't there a function for this somewhere in `cf-check`? We call functions from there in case of DB repair, etc. so I'd expect there to be something.